src/controllers/users.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Header,
+  Patch,
+  Path,
+  Post,
+  Request,
+  Route,
+  Security,
+  Tags
+} from 'tsoa'
+import bcrypt from 'bcryptjs'
+import { db } from '../util/database'
+import { User } from '../models/users/user'
+import { PostUsersRequest } from '../models/users/postUsersRequest'
+import { PatchUsersRequest } from '../models/users/patchUsersRequest'
+import { Role } from '../enums/role'
+import Error403 from '../errors/error403'
+import Error404 from '../errors/error404'
+import Error409 from '../errors/error409'
+
+@Tags('users')
+@Route('users')
+export class Users extends Controller {
+  @Get()
+  @Security('jwt', [Role.Admin, Role.User])
+  public async getUsers(): Promise<User[]> {
+    const result = await db()
+      .query(`SELECT u.id, u.email, r.name AS role, u.xmin AS transactionId
+      FROM users u
+      JOIN roles r on u.roleId = r.id 
+      WHERE u.deleteDate IS null`)
+
+    return result.rows.map((r) => {
+      return <User>{
+        id: r.id,
+        email: r.email,
+        role: r.role
+      }
+    })
+  }
+
+  @Get(':id')
+  @Security('jwt', [Role.Admin, Role.User])
+  public async getUser(id: number): Promise<User> {
+    const result = await db().query(
```

Comment:
You shouldn't keep DB query into controller. Move this code into repository / dao.

Reply:
done